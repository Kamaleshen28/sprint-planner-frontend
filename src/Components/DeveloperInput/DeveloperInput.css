.dev-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  /* align-content: center; */
  /* min-height: 57vh; */
  /* width: 100vw; */
  margin: 0 10px 0 10px;
  border-radius: 2px;
  background-color: white;
  box-shadow: 0px 0px 11px -8px #000000;
  padding: 0 20px 0 20px;
}

.developer-container-form {
  width: 100%;
  /* height: 15vh; */
  display: flex;
  flex-direction: column;
  background-color: white;
  justify-content: center;
  align-items: center;
  border-radius: 2px;
  box-shadow: 0px 0px 11px -8px #000000;
  /* margin-top:30px; */
  min-height: 60px;
}
.dev-container form {
  width: 100%;
  display: flex;
  flex-direction: column;
  height: 80px;
  margin-bottom: 20px;
  /* align-content: space-between; */
  /* justify-content: space-around; */
}

.dev-list {
  max-height: 50vh;
  /* border-radius: 20px; */
  /* box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.2); */
  /* border: 1px solid lightblue; */
  /* add scroll */
  overflow-y: scroll;
  margin-bottom: 10px;
}
.dev-list::-webkit-scrollbar {
  display: none;
}
.dev-list:hover::-webkit-scrollbar {
  display: block;
}
