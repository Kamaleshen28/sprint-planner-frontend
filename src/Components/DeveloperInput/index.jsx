import { Box, Input, Fab, Tooltip, Modal, Button } from '@mui/material';
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';
import React, { useState } from 'react';
import DeveloperEntry from '../DeveloperEntry';
import './DeveloperInput.css';

function Item(props) {
  const { sx, ...other } = props;
  return (
    <Box
      sx={{
        p: 1,
        m: 1,
        width: 200,
        bgcolor: (theme) =>
          theme.palette.mode === 'dark' ? '#101010' : 'grey.100',
        color: (theme) =>
          theme.palette.mode === 'dark' ? 'grey.300' : 'grey.800',
        border: '1px solid',
        borderColor: (theme) =>
          theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
        borderRadius: 2,
        fontSize: '0.875rem',
        fontWeight: '700',
        ...sx,
        display: 'flex',
        alignItems: 'center',
        // alignSelf: 'center',
      }}
      {...other}
    />
  );
}

Item.propTypes = {
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool]),
    ),
    PropTypes.func,
    PropTypes.object,
  ]),
};

export default function DeveloperInput({
  developerList,
  setDeveloperList,
  deleteCheck,
}) {
  const [id, setId] = useState(
    developerList.length > 0
      ? developerList[developerList.length - 1].id + 1
      : 1,
  );
  const [developer, setDeveloper] = useState('');
  const [sprintCapacity, setSprintCapacity] = useState(null);
  const [capacity, setCapacity] = useState(null);

  const removeItem = (id) => {
    let newDeveloperList = developerList.filter(
      (developer) => developer.id !== id,
    );
    setDeveloperList(newDeveloperList);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (id && developer && capacity && sprintCapacity) {
      const newDeveloper = { id, developer, sprintCapacity, capacity };
      setDeveloperList((developerList) => {
        return [...developerList, newDeveloper];
      });
      setId(id + 1);
      setDeveloper('');
    }
  };
  //   console.log(developerList.length);
  return (
    <div className="dev-container">
      <Box mx="auto">
        <div>
          <Box
            sx={{
              display: 'grid',
              gridTemplateColumns: '1fr 2fr repeat(3, 1fr)',
              p: 1,
              ml: 6,
              // m: 1,
              // justifyContent: 'space-between',
              // textAlign: 'center',
              // bgcolor: 'background.white',
              // borderRadius: 1,
              // gap: '0 1rem',
            }}
          >
            <Item sx={{ width: '80%', mx: 'auto' }}>Serial No.</Item>
            <Item sx={{ width: '80%', mx: 'auto' }}>Developers</Item>
            <Item sx={{ width: '80%', mx: 'auto' }}>Sprint Capacity(days)</Item>
            <Item sx={{ width: '80%', mx: 'auto' }}>Capacity(days)</Item>
          </Box>
          <div className="dev-list">
            {developerList.length === 0 ? (
              <p style={{ fontSize: '2rem' }}>No Developers</p>
            ) : (
              developerList.map((developerInfo, index) => {
                return (
                  <DeveloperEntry
                    key={developerInfo.id}
                    index={index}
                    developerInfo={developerInfo}
                    removeItem={removeItem}
                    deleteCheck={deleteCheck}
                  />
                );
              })
            )}
          </div>
        </div>
        {/* <form onSubmit={handleSubmit}> */}
        <ValidatorForm
          onSubmit={handleSubmit}
          onError={(errors) => console.log(errors)}
        >
          <Box
            sx={{
              // display: 'flex',
              // flexDirection: 'row',
              display: 'grid',
              gridTemplateColumns: '1fr 2fr repeat(3, 1fr)',

              // ml: 6,
              // p: 1,
              // p: '0 2',
              // m: '0 4',
              // justifyContent: 'space-around',
              bgcolor: '#28709e',
              borderRadius: '0 0 20px 20px',
              height: '100px',
              // boxShadow: 1,
              alignItems: 'center',
              // height: '100%',
            }}
          >
            {/* <Item sx={{ width: '80%', mx: 'auto' }}>
              <TextValidator
                placeholder="Id"
                type="number"
                name="id"
                disabled
                value={id}
              />
            </Item> */}
            <div></div>
            <Item sx={{ width: '80%', mx: 'auto' }}>
              <TextValidator
                style={{ width: '100%' }}
                placeholder="Enter Developer Name"
                type="text"
                name="developer"
                defaultValue="Disabled"
                value={developer}
                validators={['required', 'matchRegexp:^[a-zA-Z ]*$']}
                errorMessages={['this field is required', 'Not a valid name']}
                onChange={(e) => setDeveloper(e.target.value)}
              />
            </Item>
            <Item sx={{ width: '80%', mx: 'auto' }}>
              <TextValidator
                placeholder="Enter Sprint Capacity"
                type="number"
                name="sprintCapacity"
                value={sprintCapacity}
                validators={['required', 'isNumber', 'minNumber:0']}
                errorMessages={[
                  'this field is required',
                  'Not a number',
                  'Should be positive',
                ]}
                onChange={(e) => setSprintCapacity(e.target.value)}
              />
            </Item>
            <Item sx={{ width: '80%', mx: 'auto' }}>
              <TextValidator
                placeholder="Enter capacity"
                type="number"
                name="capacity"
                value={capacity}
                validators={['required', 'isNumber', 'minNumber:0']}
                errorMessages={[
                  'this field is required',
                  'Not a number',
                  'Should be positive',
                ]}
                onChange={(e) => setCapacity(e.target.value)}
              />
            </Item>
            <Tooltip sx={{ mx: 'auto' }} title="Add Developer" aria-label="add">
              <Fab color="primary" type="submit" aria-label="add">
                +
              </Fab>
            </Tooltip>
          </Box>
        </ValidatorForm>
        {/* </form> */}
      </Box>
    </div>
  );
}

DeveloperInput.propTypes = {
  developerList: PropTypes.array,
  setDeveloperList: PropTypes.func,
  deleteCheck: PropTypes.func,
};
