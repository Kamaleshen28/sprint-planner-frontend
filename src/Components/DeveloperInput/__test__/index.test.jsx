import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import DeveloperInput from '..';

describe('DeveloperInput', () => {
  it('should render correctly', () => {
    const mockProps = {
      developerList: [],
      setDeveloperList: jest.fn(),
      deleteCheck: jest.fn(),
    };
    const { asFragment } = render(<DeveloperInput {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
