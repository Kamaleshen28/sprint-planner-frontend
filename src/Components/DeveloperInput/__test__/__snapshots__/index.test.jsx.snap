// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeveloperInput should render correctly 1`] = `
<DocumentFragment>
  <div
    class="dev-container"
  >
    <div
      class="MuiBox-root css-ibnnmb"
    >
      <div>
        <div
          class="MuiBox-root css-gp3o3r"
        >
          <div
            class="MuiBox-root css-1ur5le0"
          >
            Serial No.
          </div>
          <div
            class="MuiBox-root css-1ur5le0"
          >
            Developers
          </div>
          <div
            class="MuiBox-root css-1ur5le0"
          >
            Sprint Capacity(days)
          </div>
          <div
            class="MuiBox-root css-1ur5le0"
          >
            Capacity(days)
          </div>
        </div>
        <div
          class="dev-list"
        >
          <p
            style="font-size: 2rem;"
          >
            No Developers
          </p>
        </div>
      </div>
      <form
        class="developer-container-form"
      >
        <div
          class="MuiBox-root css-1c6mtae"
        >
          <div />
          <div
            class="MuiBox-root css-1ur5le0"
          >
            <div>
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                style="width: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    name="developer"
                    placeholder="Enter Developer Name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1ur5le0"
          >
            <div>
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r1:"
                    name="sprintCapacity"
                    placeholder="Enter Sprint Capacity"
                    type="number"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1ur5le0"
          >
            <div>
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    name="capacity"
                    placeholder="Enter capacity"
                    type="number"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            aria-label="add"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-1ioynfo-MuiButtonBase-root-MuiFab-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="submit"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
